pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -100,51 +103,184 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
-	// Define a new Pod object
-	pod := newPodForCR(instance)
+	secret := newPostgresSecret(instance)
 
 	// Set UnifiedPushServer instance as the owner and controller
-	if err := controllerutil.SetControllerReference(instance, pod, r.scheme); err != nil {
+	if err := controllerutil.SetControllerReference(instance, secret, r.scheme); err != nil {
 		return reconcile.Result{}, err
 	}
 
-	// Check if this Pod already exists
-	found := &corev1.Pod{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
+	// Check if this Deployment already exists
+	foundSecret := &corev1.Secret{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: secret.Name, Namespace: secret.Namespace}, foundSecret)
 	if err != nil && errors.IsNotFound(err) {
-		reqLogger.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
-		err = r.client.Create(context.TODO(), pod)
+		reqLogger.Info("Creating a new Secret", "Secret.Namespace", secret.Namespace, "Secret.Name", secret.Name)
+		err = r.client.Create(context.TODO(), secret)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Define a new Deployment object
+	deployment := newDeploymentForCR(instance)
```

Comment:
I'm guessing it's fine that an operator creates a deployment and watches that. It could alternatively create the pods (or ReplicaSets) directly and manage those itself (like how ReplicaSet (or Deployment) controllers already do. I'm not sure which is better?

Reply:
One of the difference between Deployment and ReplicaSet is to Deployment can control how rolling updates are performed. So if you use Deployment, you can configure it there, otherwise you will need to implement the updates in your code too?

Reply:
Yeah, it seems natural to use Deployment, but I guess I was just doubting myself, given that the operator-sdk default creates a Pod. I'm sure there are some other cases where actually doing it all in the operator gives people more flexibility or something, but we probably don't need that